src/Phinx/Db/Adapter/SQLiteAdapter.php

```diff
@@ -846,7 +846,15 @@ protected function copyAndDropTmpTable(AlterInstructions $instructions, string $
                 )
             );
 
+            $foreignKeysEnabled = (bool)$this->fetchRow('PRAGMA foreign_keys')['foreign_keys'];
+            if ($foreignKeysEnabled) {
+                $this->execute('PRAGMA foreign_keys = OFF');
+            }
             $this->execute(sprintf('DROP TABLE %s', $this->quoteTableName($tableName)));
+            if ($foreignKeysEnabled) {
+                $this->execute('PRAGMA foreign_keys = ON');
```

Comment:
Does toggling back on foreign key pragma also revalidate any existing FKs, or do we need to also run `PRAGMA foreign_key_check;`?

Reply:
AFAICT it doesn't (and apparently neither will renaming the temp table). But is a revalidation at that point really neccessary? Do we expect `copyDataToNewTable()` to possibly silently fail (in whatever way)?

Reply:
I'd say it wouldn't hurt to do as an extra precaution though I cannot think of a case off-hand that might cause that to error.

Reply:
I'll give it a try.

Reply:
I gave it a try, and it seems to work OK, but it appears that it would introduce a breaking change for at least `changePrimaryKey()`.

Currently when the primary key is part of a foreign key constraint, and the primary key is being changed using `changePrimaryKey()`, things will just continue to work, the PK changes, and the FK constraint stays as-is. However this can actually leave the FK constraint in an invalid state, as its parent key columns must be either primary keys, or subject to a unique constraint.

Since PKs are already unique constraints in and of themselves, one would usually not put an additional unique index on top, which means that a PK change could then leave the old column without any unique constraint, and then `PRAGMA foreign_key_check()` would cause a "foreign key mismatch" SQL error.

I guess that behavior could actually be desired, as one would usually like to know when making changes that leaves the DB in an invalid state, but would that actually be in accordance with the backwards compatibility policy of this project?

Reply:
Pushed the changes so that you have something to look at.

I'm not overly happy with the need for the option whether to validate foreign keys, but there needs to be some form of control over it when running multi-step operations like `getChangePrimaryKeyInstructions()`, otherwise, in this specific case, a constraint error would be triggered when validation occurs in between removing the old PK and adding the new one.

Reply:
@MasterOdin Do you like what you see, or nah?

@dereuromark In light of [the FK problems](https://github.com/cakephp/phinx/issues/2154) you've brought up, do you possibly have any input on the BC break here? Both of this currently works, and would trigger an error after this change in case there isn't already a unique index on the current PK:

```php
 $table
    ->changePrimaryKey('other');
```
```php
 $table
    ->changePrimaryKey('other')
    ->addIndex('current', ['unique' => true]);
```

Adding the index first will continue to work:

```php
 $table
    ->addIndex('current', ['unique' => true])
    ->changePrimaryKey('other');
```

Reply:
Is this still an issue given the feature flags being added now?

Reply:
I guess it depends on whether the foreign key check is something that should be hidden behind a feature flag? And on whether this breaking change is actually considered a problem.

Reply:
I feel like this is a bugfix and that the prior behavior was incorrect as it could leave the DB in a non-valid state, and so I'm not sure I'd consider this a "breaking change". As such, I don't think we need to wrap this in a feature flag.